packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/annotations/PersistedName.kt

```diff
@@ -17,7 +17,7 @@
 package io.realm.kotlin.types.annotations
 
 @Retention(AnnotationRetention.SOURCE)
-@Target(AnnotationTarget.FIELD)
+@Target(AnnotationTarget.FIELD, AnnotationTarget.PROPERTY)
```

Comment:
not sure if this is needed actually, I was originally considering a different scenario where the backlinks child has a persisted name rather than parent

Reply:
I was looking a bit for a description between what is a field and what is a property. I found this discussion here: https://discuss.kotlinlang.org/t/property-vs-field/8179/12

And after some experiments, I think we only want to apply this annotation to `AnnotationTarget.FIELD` which is the field backing a property and the one we store in Realm.

E.g. you can have properties without backing fields (which Realm cannot persist), e.g.:

```
    // AnnotationTarget.FIELD cannot be applied to this
    var name
        get() = "foo"
        set(value) {
            _other_field = value
        }
```

Reply:
I am not sure if I understood this correctly but would it not actually imply we would still be targeting properties like @clementetb suggests? I think the backing field is not important here because the annotation is applied by the user onto the original property in the Realm object rather than its backing field. Also for cases like backlinks like
```
    @PersistedName(name = "persistedNameParent")
    val publicNameParent by backlinks(Sample::publicNameChild)
```
would error if we only target fields

Reply:
Hmm, yeah, I forgot about backlinks. True. In that case, supporting just properties are fine.